mgmtsystem/hawkular.py

```diff
@@ -26,13 +27,52 @@
 
 Feed = namedtuple('Feed', ['id', 'name', 'path'])
 ResourceType = namedtuple('ResourceType', ['id', 'name', 'path'])
-Server = namedtuple('Server', ['id', 'name', 'path'])
+Resource = namedtuple('Resource', ['id', 'name', 'path'])
+ResourceData = namedtuple('ResourceData', ['name', 'path', 'value'])
+Server = namedtuple('Server', ['id', 'name', 'path', 'data'])
 Deployment = namedtuple('Deployment', ['id', 'name', 'path'])
-ServerStatus = namedtuple('ServerStatus', ['address', 'version', 'state', 'product', 'host'])
 Event = namedtuple('event', ['id', 'eventType', 'ctime', 'dataSource', 'dataId',
                              'category', 'text'])
 
 
+class Path(object):
+    """Path class
+
+    Path is class to split canonical path to friendly values.
+
+    Args:
+        path:   The canonical path. Example: /t;28026b36-8fe4-4332-84c8-524e173a68bf\
+        /f;88db6b41-09fd-4993-8507-4a98f25c3a6b/r;Local~~
+
+    """
+    def __init__(self, path):
+        self.path = {}
+        self.path.update({'raw': path})
+        if self.path['raw']:
+            _pname_mapping = {
+                '/t;': 'tenant',
+                '/e;': 'environment',
+                '/rt;': 'resource_type',
+                '/mt;': 'metric_type',
+                '/f;': 'feed',
+                '/ot;': 'operation_type',
+                '/mp;': 'metadata_pack',
+                '/r;': 'resource',
+                '/d;': 'data',
+                '/rl;': 'relationship',
+            }
+            raw_paths = re.split(r'(/\w+;)', self.path['raw'])
+            if len(raw_paths) % 2 == 1:
+                del raw_paths[0]
+            for p_index in range(0, len(raw_paths), 2):
+                self.path.update({_pname_mapping[raw_paths[p_index]]: raw_paths[p_index + 1]})
+
+    def __getattr__(self, name):
+        return self.path[name] if name in self.path else None
```

Comment:
I wonder if it is not better to put the things in the object directly rather than have getattr do magic. You could replace the `self.path` with using the inner dictionary of the object (``self.__dict__``) and that way you will be always able to see what is available using dir().